### DISCUSSION TITLE
How to make my code work with pymodbus 3.0.2

### DISCUSSION THREAD

**Discussion by azkikr316** [pymodbus-dev/pymodbus#1179] [Help] [answered]
Hi,

Good day

The below code works on pymodbus 2.5.3 and python 3.7.
Need your help in making this code work on pymodbus3.0.2.

*****code_pymodbus2.5.3/python3.7*****

import pymodbus
from pymodbus.pdu import ModbusRequest
from pymodbus.client.sync import ModbusSerialClient as ModbusClient #initialize a serial RTU
from pymodbus.transaction import ModbusRtuFramer
#Payload Decoder
from pymodbus.constants import Endian
from pymodbus.payload import BinaryPayloadDecoder
from pymodbus.payload import BinaryPayloadBuilder
from pymodbus.compat import iteritems
from collections import OrderedDict
import logging

#Establishing RS485 connection
client = ModbusClient(method = "rtu", port="/dev/ttyUSB0", stopbits = 1, bytesize = 8, parity = 'N', baudrate= 9600 , timeout=5)
client.connect()

#Encoder - 16bit int
builder = BinaryPayloadBuilder(byteorder=Endian.Big,
                              wordorder=Endian.Little)
payload = builder.to_registers()
builder.add_16bit_int(-0x5678)
payload = builder.build()
count = len(payload)

try:
    while True:
        try: #Modbus instance (pymodbus V2.5.3 and python3.7)
            #Address STBD 0x02 (40003)
            #Debugging script
            logging.basicConfig()
            log = logging.getLogger()
            log.setLevel(logging.DEBUG)
            request = client.read_holding_registers(0x02,count,unit=0x01)
            decoder = BinaryPayloadDecoder.fromRegisters(request.registers, byteorder=Endian.Big)
            K0 = int(decoder.decode_16bit_int()) #converts to integer(int)
            print ("Modbus_Value, {0}, mA".format((K0)))
            client.close()
            sleep(5)

        except:
            pass

#except KeyboardInterrupt:
except (KeyboardInterrupt, SystemExit, Exception):
    client.close()

**Comment by janiversen** [comment, marked as answer]
Please read the documentation and see our examples.
